controllers/etcd_controller.go

```diff
@@ -179,11 +180,29 @@ func (r *EtcdReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	if !reflect.DeepEqual(etcd.Spec, etcdCopy.Spec) {
 		etcdCopy.Spec = etcd.Spec
 	}
-	logger.Infof("Reconciling etcd: %s/%s", etcd.GetNamespace(), etcd.GetName())
+
+	etcdCopy.Status.ObservedGeneration = &etcdCopy.Generation
+	if _, ok := etcd.Annotations[common.GardenerOperation]; ok {
```

Comment:
Shouldn't you work on the `etcdCopy` from here on?

Reply:
done